### PR TITLE
EUI-3825: Missing FTPA tab

### PR DESCRIPTION
**Pull request by paul-graham**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3825 ✅ `DONE`

### Change description ###
Fixed an issue where fields on one tab weren't shown if they depended on fields from another tab. This was visible on the FTPA tab within the IA jurisdiction but was actually a generic issue. The solution was to essentially the same as one that was put in place for the **Check your answers** page.

Some of the changes are simply cleaning up the code within the `CaseViewerComponent` - something we're doing as we go.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/components/case-viewer/case-viewer.component.ts [hunk 1]
```diff
@@ -34,44 +44,45 @@ export class CaseViewerComponent implements OnInit, OnDestroy, AfterViewInit {
...
+    private readonly ngZone: NgZone,
+    private readonly route: ActivatedRoute,
+    private readonly navigationNotifierService: NavigationNotifierService,
+    private readonly orderService: OrderService,
+    private readonly activityPollingService: ActivityPollingService,
+    private readonly dialog: MatDialog,
+    private readonly alertService: AlertService,
+    private readonly draftService: DraftService,
+    private readonly caseNotifier: CaseNotifier,
+    private readonly errorNotifierService: ErrorNotifierService,
```
Comment (phillip-whitaker) [inline src/shared/components/case-viewer/case-viewer.component.ts:80]: Like to see this :)

File: src/shared/components/case-viewer/case-viewer.component.ts [hunk 2]
```diff
@@ -34,44 +44,45 @@ export class CaseViewerComponent implements OnInit, OnDestroy, AfterViewInit {
...
     private readonly location: Location
   ) {
   }
 
-  ngOnInit() {
+  public ngOnInit(): void {
```
Comment (phillip-whitaker) [inline src/shared/components/case-viewer/case-viewer.component.ts:85]: I am not really a fan of putting accessor declarations on implemented methods, it goes against the standard Angular coding style, any reason for this?

File: src/shared/components/case-viewer/case-viewer.component.ts [hunk 3]
```diff
@@ -94,11 +105,11 @@ export class CaseViewerComponent implements OnInit, OnDestroy, AfterViewInit {
     });
   }
 
-  isPrintEnabled(): boolean {
+  public isPrintEnabled(): boolean {
     return this.caseDetails.case_type.printEnabled;
   }
 
-  ngOnDestroy() {
+  public ngOnDestroy(): void {
```
Comment (phillip-whitaker) [inline src/shared/components/case-viewer/case-viewer.component.ts:112]: same comment here re:implemented methods public keyword

File: src/shared/components/case-viewer/case-viewer.component.ts [hunk 4]
```diff
@@ -207,17 +218,14 @@ export class CaseViewerComponent implements OnInit, OnDestroy, AfterViewInit {
...
     this.sortedTabs = this.orderService.sort(this.caseDetails.tabs);
-
     this.caseFields = this.getTabFields();
-
-    console.log('caseDetails', this.caseDetails);
-
     this.sortedTabs = this.sortTabFieldsAndFilterTabs(this.sortedTabs);
 
-    console.log('sortedTabs', this.sortedTabs)
+    // EUI-3825. Build a FormGroup for use in the filtering.
```
Comment (phillip-whitaker) [inline src/shared/components/case-viewer/case-viewer.component.ts:227]: Does this comment add value? You've added a JsDoc to the method which covers it

**Review by kuda1992 — DISMISSED** [review]
Looks good

**Review by phillip-whitaker — COMMENTED** [review]
Couple of questions etc.

File: src/shared/components/case-viewer/case-viewer.component.ts (hunk 1 above)
Comment (phillip-whitaker) [reply src/shared/components/case-viewer/case-viewer.component.ts:80]: brownie points ++

File: src/shared/components/case-viewer/case-viewer.component.ts (hunk 2 above)
Comment (paul-graham) [reply src/shared/components/case-viewer/case-viewer.component.ts:85]: My understanding is that putting _explicit_ `public` accessors on implicitly public methods - including these ones - is part of the `tslint` standard. We have a bit of a disparity between MC and the toolkit in this regard but MC flags it as a warning and I think we should be consistent across all the projects we're responsible for. 🤷‍♂️

Comment (paul-graham) [reply src/shared/components/case-viewer/case-viewer.component.ts:85]: Where is the Angular coding style guide that says we shouldn't do that? Happy to be proven wrong on this one but I think `tslint` flags it as a warning within MC if it's missing...

File: src/shared/components/case-viewer/case-viewer.component.ts (hunk 3 above)
Comment (paul-graham) [reply src/shared/components/case-viewer/case-viewer.component.ts:112]: Same as above. :)

File: src/shared/components/case-viewer/case-viewer.component.ts (hunk 4 above)
Comment (paul-graham) [reply src/shared/components/case-viewer/case-viewer.component.ts:227]: Good call. This was my initial comment before I added the JsDoc. :)

**Review by phillip-whitaker — APPROVED** [review]

File: src/shared/components/case-viewer/case-viewer.component.ts (hunk 4 above)
Comment (paul-graham) [reply src/shared/components/case-viewer/case-viewer.component.ts]: Done. 👍

File: src/shared/components/case-viewer/case-viewer.component.ts (hunk 2 above)
Comment (paul-graham) [reply src/shared/components/case-viewer/case-viewer.component.ts]: Reverted, pending a conversation in the Tech Forum about what we _should_ be doing as there's enormous inconsistency across within each project and across all the projects.

File: src/shared/components/case-viewer/case-viewer.component.ts (hunk 3 above)
Comment (paul-graham) [reply src/shared/components/case-viewer/case-viewer.component.ts]: Reverted, pending a conversation in the Tech Forum about what we _should_ be doing as there's enormous inconsistency across within each project and across all the projects.

**Review by phillip-whitaker — APPROVED** [review]
All looks good and comments replied to an agreed.

**Review by kuda1992 — APPROVED** [review]
looks good

**Review by Philyorkshire — APPROVED** [review]
LGTMp

**Pull request merged**